agent/acs/handler/acs_handler.go

```diff
@@ -245,33 +245,45 @@ func (acsSession *session) Start() error {
 			} else if cfg.DisconnectCapable.Enabled() {
 				// Check if this instance has ECS_DISCONNECT_CAPABLE turned on. This code block is only reached when
 				// ACS has already failed to connect.
-				if acsSession.disconnectionTimer != nil {
-					timerCompleted := acsSession.checkDisconnectionTimer()
-					logger.Debug("Checking disconnectionTimer status", logger.Fields{
-						"timerCompleted": timerCompleted,
-					})
-					if timerCompleted {
-						// If the timer has been completed, then set DisconnectModeEnabled to true.
-						cfg.SetDisconnectModeEnabled(true)
-						logger.Debug("Turning DisconnectModeEnabled on after timer is completed", logger.Fields{
-							"disconnectionMode": cfg.GetDisconnectModeEnabled(),
-						})
-						acsSession.disconnectionTimer = nil
+				switch cfg.GetDisconnectModeEnabled() {
```

Comment:
Just to make sure my understanding of current re-connection strategy.
1. 1st loop: When `cfg.DisconnectCapable.Enabled() is true` -> Enter `cfg.GetDisconnectModeEnabled() is false` block -> Enter line 283-285 as `acsSession.disconnectionTimer is nil`
2.  2nd loop: `cfg.DisconnectCapable.Enabled() is true` -> Enter `cfg.GetDisconnectModeEnabled() is false` block -> Enter line 259 as `acsSession.disconnectionTimer is not nil` -> Enter line 273-280 to wait for `reconnectToACSTimeout`
3. 3rd loop: `cfg.DisconnectCapable.Enabled() is true` -> Enter `cfg.GetDisconnectModeEnabled() is false` block -> Enter line 259 as `acsSession.disconnectionTimer is not nil` -> Enter line 264-271 when `timerCompleted`
4. 4th loop: `cfg.DisconnectCapable.Enabled() is true` -> Enter `cfg.GetDisconnectModeEnabled() is true` block -> Enter line 250-257 to wait for `disconnectTimeout` before starting the new attempt to reconnect to ACS
5. 5th and so on loop: keep reconnecting to ACS every `disconnectTimeout` interval time until the connection can be setup

Please correct me if it is not correct, add some details and benefits on PR's overview to explain why we want to add this new timer. Thanks!

Reply:
I used the switch case to try and reduce the number of if/else's in this function

Reply:
@Realmonia I just wanted to clarify, do you mean something like moving lines 260-281 into their own separate function (without changing any of the logic) and calling that function in line 260 to reduce the levels of conditionals? Thanks!

Reply:
@rsheik29 yep - starting from 258 would be fine too. And maybe change from switch/case to if/else consider they are not really different